4-auth/structured_data/active_record/app/models/book.rb

```diff
@@ -23,15 +36,13 @@ class Book < ActiveRecord::Base
   after_create :upload_image, if: :cover_image
 
   def upload_image
-    image = StorageBucket.files.new(
-      key: "cover_images/#{id}/#{cover_image.original_filename}",
-      body: cover_image.read,
-      public: true
-    )
-
-    image.save
+    file = Book.storage_bucket.create_file \
```

Comment:
Apply fix to other related files as well. Please move up `cover_image.tempfile` instead of using `\`.

Reply:
That causes Rubocop to complain because you can't line up the parameters without going over the line limit. This is the recommended alternative (or alternately, use parens).

Reply:
Sg, move forward with `\`.